src/main/java/de/tum/cit/aet/artemis/text/web/TextSubmissionResource.java

```diff
@@ -166,11 +170,14 @@ private ResponseEntity<TextSubmission> handleTextSubmission(long exerciseId, Tex
     @EnforceAtLeastStudent
     public ResponseEntity<TextSubmission> getTextSubmissionWithResults(@PathVariable long submissionId) {
         log.debug("REST request to get text submission: {}", submissionId);
-        var textSubmission = textSubmissionRepository.findWithEagerResultsAssessorById(submissionId).orElseThrow(() -> new EntityNotFoundException("TextSubmission", submissionId));
-
-        if (!authCheckService.isAtLeastTeachingAssistantForExercise(textSubmission.getParticipation().getExercise())) {
-            // anonymize and throw exception if not authorized to view submission
-            plagiarismService.checkAccessAndAnonymizeSubmissionForStudent(textSubmission, userRepository.getUser().getLogin(), textSubmission.getParticipation());
+        var textSubmission = textSubmissionRepository.findWithEagerResultsAssessorByIdElseThrow(submissionId);
```

Comment:
decided to keep the access checks in this resource, bcs. checks for text seems to be different here to other exercise types. there is no such endpoint for the other exercise types